### PR TITLE
ENT-278: Allow users with exotic characters in email addresses to be unlinked

### PR DESCRIPTION
**Pull request by haikuginger**
**Description:** Previously, a failure to urlencode users' email addresses when submitting them to be unlinked from an enterprise customer meant that users with characters like `+` in their email addresses could not be unlinked. This pull request ensures that email addresses are urlencoded prior to their submission.

**JIRA:** [ENT-278](https://openedx.atlassian.net/browse/ENT-278)

**Dependencies:** None

**Merge deadline:** 10 April 2017

**Testing instructions:**

1. With the existing code base, link a user with an email address that includes `+`.
2. In the Django admin Manage Learners page, attempt to unlink that learner. Observe that it fails, and that the error message produced replaces the `+` with a ` `.
3. Pull this code.
4. Again attempt to unlink the same user.
5. Observe that the process succeeds.

**Merge checklist:**

- [x] Regenerate requirements with `make upgrade && make requirements` (and make sure to fix any errors).
  **DO NOT** just add dependencies to `requirements/*.txt` files.
- [x] All reviewers approved
- [x] CI build is green
- [x] Version bumped
- [x] Changelog record added
- [x] Documentation updated (not only docstrings)
- [x] Commits are (reasonably) squashed
- [x] Translations are updated
- [x] PR author is listed in AUTHORS

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is finished.
- [ ] Delete working branch (if not needed anymore)

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @haikuginger! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

<a   href="https://openedx-webhooks.herokuapp.com/github/process_pr?number=78&repo=edx%2Fedx-enterprise"   rel="nofollow" >Create an OSPR issue for this pull request</a>.

**Comment by mattdrayer** [comment]
Assuming testing with the `+` character is enough for us to consider in the way of funky-but-acceptable characters found in email addresses, I am 👍 as well.

**Comment by haikuginger** [comment]
@mattdrayer, we also hit the `@` sign. ;)

**Comment by mattdrayer** [comment]
💯

File: spec/javascripts/fixtures/manage_learners_form.html [hunk 1]
```diff
@@ -57,3 +57,11 @@
       </div>
   </div>
 </div>
+<table class="learners-table linked-learners">
+  <tr class="form-row" data-email="escaped+email@google.com">
+    <td class="email-container">
+      escaped+email@google.com
+    </td>
+    <td class="delete-cell"><input type="button" value="Unlink" id="learner-unlink-button"></td>
```
Comment (itsjeyd) [inline spec/javascripts/fixtures/manage_learners_form.html]: @haikuginger Minor nit: Formatting. The following would be even more readable:

```html
    <td class="delete-cell">
      <input type="button" value="Unlink" id="learner-unlink-button">
    </td>
```

**Comment by haikuginger** [comment]
Checklist from 86742aa.

**Comment by haikuginger** [comment]
Woo! :shipit:

**Pull request merged**